### ISSUE TITLE
Do not filter __* fields

### ISSUE THREAD

**Issue by comigor** [comigor/artemis#41]
_Originally posted by @michael-golfi in https://github.com/comigor/artemis/issues/31_

I use `__typename/id` as a key for client side caching.

It's currently, on purpose, filtering any field that starts with `__`.

**Comment by klavs** [comment]
@comigor any leading `_` could be changed to `$` in Dart as names cannot contain `$` in GraphQL.

https://graphql.github.io/graphql-spec/June2018/#sec-Names

**Comment by smkhalsa** [comment]
I have the same issue as @michael-golfi. I plan to use `__typename` (with `id`) for client side caching. @klavs's suggestion of converting `__` to `$` makes sense to me.

@comigor what are your thoughts on this?

@michael-golfi, have you already built your caching layer? I need a normalized, offline-first cache and I'd love to collaborate on an existing project if one exists.

**Comment by comigor** [comment]
Yes, it seems a good aproach as (of course) underscore fields would be privatized in Dart.

**Comment by michael-golfi** [comment]
@smkhalsa I used [graphql-flutter](https://github.com/zino-app/graphql-flutter/tree/master/packages/graphql_flutter#offline-cache)

Getting back into this after the holidays, has there been any movement on this?

**Comment by vasilich6107** [comment]
Hi @michael-golfi 
Could you try this branch while it awaits the review?
```
artemis:
    git:
      url: https://github.com/comigor/artemis.git
      ref: bugfix/underscored-items-another-solution
```

**Comment by michael-golfi** [comment]
Hi @vasilich6107 

I see the following property now on query:

```dart
  @override
  @JsonKey(name: '__typename')
  String $$typename;
```

It looks good so far and I would have to let you know when I start using the classes in my project.

**Comment by vasilich6107** [comment]
Closed by #142
